sips/voluntary_exit.md

```diff
@@ -53,4 +53,227 @@ type SignedExitMessage struct {
 A new runner for voluntary exits will be created.
 
 Upon exit message being parsed from EL, each node taking part in the specific cluster will broadcast SignedExitMessage including a partial signature for the voluntary exit.
-Upon a quorum of partial signatures, each node will reconstruct a valid signature and broadcast it to the beacon chain
\ No newline at end of file
+Upon a quorum of partial signatures, each node will reconstruct a valid signature and broadcast it to the beacon chain
+
+
+# Proposed solution
+
+_Validator voluntary exit_ will be a new type of duty. It'll be similar to validator registration with no consensus and no post-consensus steps (just pre-consensus).
+
+For this duty, operators need to create a [`SignedVoluntaryExit`](https://github.com/attestantio/go-eth2-client/blob/master/spec/phase0/signedvoluntaryexit.go#L28):
+```go
+type SignedVoluntaryExit struct {
+	Message   *VoluntaryExit
+	Signature BLSSignature `ssz-size:"96"`
+}
+```
+where _Signature_ is the validator's signature over [`VoluntaryExit`](https://github.com/attestantio/go-eth2-client/blob/master/spec/phase0/voluntaryexit.go#L27), defined as:
+```go
+type VoluntaryExit struct {
+	Epoch          Epoch
+	ValidatorIndex ValidatorIndex
+}
+type Epoch uint64
+type ValidatorIndex uint64
+```
+
+This requires the following changes:
+- New [_VoluntaryExitCall_ on BeaconNode](#beacon-node) interface.
+- New [_PartialSigMsgType_](#partial-signature-message-type).
+- New [_BeaconRole_](#beacon-role).
+- New [Runner struct implementation](#runner) .
+
+
+
+## Beacon Node
+
+New call to be included to the `BeaconNode` interface (according to [attestantio/go-eth2-client:service.go](https://github.com/attestantio/go-eth2-client/blob/master/service.go#L383)).
+
+```go
+// ValidatorVoluntaryExitCalls interface has all validator voluntary exit duty specific calls
+type ValidatorVoluntaryExitCalls interface {
+    // SubmitVoluntaryExit submits a validator voluntary exit
+    SubmitVoluntaryExit(voluntaryExit *phase0.SignedVoluntaryExit, sig phase0.BLSSignature) error
+}
+
+type BeaconNode interface {
+	// GetBeaconNetwork returns the beacon network the node is on
+	GetBeaconNetwork() types.BeaconNetwork
```

Comment:
don't we have it already in the spec? why is this needeD?

Reply:
Here, I mentioned the BeaconNode interface just to show how it updates.
But we don't have yet the specification of the VoluntaryExit call.